### PR TITLE
NCL5385 Nexus plugin

### PR DESCRIPTION
**Pull request by rnc**

### CODE REVIEW

**Comment by rnc** [comment]
cc @TomasHofman

File: manipulation/src/main/java/org/jboss/gm/manipulation/ManipulationPlugin.java [hunk 1]
```diff
@@ -27,11 +25,14 @@
 import org.jboss.gm.manipulation.actions.UploadTaskTransformerAction;
 
 import static org.apache.commons.lang.StringUtils.isEmpty;
+import static org.apache.commons.lang.StringUtils.isNotEmpty;
 
 @SuppressWarnings("unused")
```
Comment (TomasHofman) [inline manipulation/src/main/java/org/jboss/gm/manipulation/ManipulationPlugin.java]: Looks like we should also remove this suppression. The only thing it hides is the non-used isDependencyManagementPluginPomCustomizationEnabled method :).

File: manipulation/src/main/java/org/jboss/gm/manipulation/ManipulationPlugin.java [hunk 2]
```diff
@@ -95,25 +96,6 @@ public void apply(Project project) {
         configurePublishingTask(configuration, project, correspondingModule, resolvedDependenciesRepository);
     }
 
-    // Ensure that if the Spring Dependency Management plugin is applied,
-    // that it's configured to not generate a "dependencyManagement" section in the generated bom
-    // This is needed because if we don't do it, the "dependencyManagement" section (which is a bom inclusion) will override our dependencies
-    // On the implementation side of things, we need to use reflection because we can get a nasty classloader errors
-    // when trying to cast the object to the known type
-    private boolean isDependencyManagementPluginPomCustomizationEnabled(Object obj) {
```
Comment (TomasHofman) [inline manipulation/src/main/java/org/jboss/gm/manipulation/ManipulationPlugin.java]: +1

File: analyzer/src/main/resources/gme-pluginconfigs.gradle [hunk 3]
```diff
@@ -18,6 +18,13 @@ if (!project.hasProperty("gmeAnalyse")) {
...
             externalDocumentationLinks.clear()
         }
+    }
 
+    allprojects {
+        project.pluginManager.withPlugin("com.bmuschko.nexus") {
+            nexus {
+                sign = false
+            }
+        }
```
Comment (TomasHofman) [inline analyzer/src/main/resources/gme-pluginconfigs.gradle]: +1

File: manipulation/src/main/java/org/jboss/gm/manipulation/ManipulationPlugin.java (hunk 1 above)
Comment (rnc) [reply manipulation/src/main/java/org/jboss/gm/manipulation/ManipulationPlugin.java]: Good catch!

File: manipulation/src/main/java/org/jboss/gm/manipulation/ManipulationPlugin.java [hunk 4]
```diff
@@ -152,10 +134,12 @@ private void configurePublishingTask(Configuration config, Project project, Mani
                     deployPlugin = MAVEN_PUBLISH_PLUGIN;
                 } else if (evaluatedProject.getPluginManager().hasPlugin(LEGACY_MAVEN_PLUGIN)) {
                     deployPlugin = LEGACY_MAVEN_PLUGIN;
+                } else if (evaluatedProject.getPluginManager().hasPlugin(LEGACY_MAVEN_PLUGIN_NEXUS)) {
+                    deployPlugin = LEGACY_MAVEN_PLUGIN_NEXUS;
```
[the remark this answers is not in the document]
Comment (rnc) [reply manipulation/src/main/java/org/jboss/gm/manipulation/ManipulationPlugin.java]: +1 agreed.

**Pull request merged**